app/models/assignment.rb

```diff
@@ -379,42 +379,125 @@ def clone_groupings_from(assignment_id)
+              # and don't return an error
+              return
+            else
+              # CASE: Group already exists with the same repo name and has a
+              #     grouping for the current assignment BUT the membership is
+              #     different.
+              # Return error since the membership is different
+              duplicate_group_error = I18n.t(
+                "csv.group_with_different_membership_current_assignment",
+                group_name: row[0])
+              return duplicate_group_error
+            end
+          end
+        end
+      end
+
     end
 
-    # Since repo_name of "group" will be set before the first save call, the
-    # set repo_name will be used instead of the autogenerated name. See
-    # set_repo_name and build_repository in the groups model. Also, see
-    # create_group_for_working_alone_student in the students model for
-    # similar semantics.
-    if is_candidate_for_setting_custom_repo_name?(row)
-      # Do this only if user_name exists and is a student.
-      if Student.where(user_name: row[2]).first
-        group.repo_name = row[0]
+    # If any of the given members do not exist or is part of another group,
+    # an error is returned without creating a group
+    unless membership_unique?(row)
+      if !errors[:groupings].blank?
+        # groupings error set if a member is already in different group
+        membership_error = I18n.t("csv.memberships_not_unique",
+                                  group_name: row[0],
+                                  student_user_name: errors.
+                                      get(:groupings).first)
+        errors.delete(:groupings)
       else
-        # Student name does not exist, use provided repo_name
-        group.repo_name = row[1]
+        # student_membership error set if a member does not exist
+        membership_error = I18n.t(
+          "csv.member_does_not_exist",
+          group_name: row[0],
+          student_user_name: errors.get(:student_memberships).first)
+        errors.delete(:student_memberships)
       end
+      return membership_error
     end
 
-    # If we are not repository admin, set the repository name as provided
-    # in the csv upload file
-    unless group.repository_admin?
-      group.repo_name = row[1]
+    # If this assignment is an individual assignment, then the repostiory
+    # name is set to be the student's user name. If this assignment is a
+    # group assignment then the repository name is taken from the csv file
+    if is_candidate_for_setting_custom_repo_name?(row)
+      repo_name = row[2]
+    else
+      repo_name = row[1]
     end
+
+    # If a repository already exists with the same repo name as the one given
+    #  in the csv file, error is returned and the group is not created
+    if repository_already_exists?(repo_name)
+      repository_error = I18n.t("csv.repository_already_exists",
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.